crates/wasmtime/src/runtime/vm/sys/unix/mmap.rs

```diff
@@ -11,6 +11,19 @@ pub struct Mmap {
     memory: SendSyncPtr<[u8]>,
 }
 
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "illumos")] {
+        // On illumos, by default, mmap reserves what it calls "swap space" ahead of time, so that
+        // memory accesses are guaranteed not to fail once mmap succeeds. NORESERVE is for cases
+        // where that memory is never meant to be accessed -- e.g. memory that's used as guard
+        // pages.
+        pub(super) const MMAP_NORESERVE_FLAG: rustix::mm::MapFlags =
+            rustix::mm::MapFlags::NORESERVE;
```

Comment:
Having spent some time digging into this, I believe that we should also pass in NORESERVE on Linux here as well. On Linux this is a bit hidden with overcommit, but e.g. Chromium passes in NORESERVE: https://chromium.googlesource.com/v8/v8/+/chromium/2840/src/base/platform/platform-linux.cc#234